datadog/fwprovider/data_source_datadog_integration_aws_iam_permissions_resource_collection.go

```diff
@@ -0,0 +1,79 @@
+package fwprovider
+
+import (
+	"context"
+
+	"github.com/DataDog/datadog-api-client-go/v2/api/datadogV2"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+
+	"github.com/terraform-providers/terraform-provider-datadog/datadog/internal/utils"
+)
+
+var (
+	_ datasource.DataSource = &awsIntegrationIAMPermissionsResourceCollectionDataSource{}
+)
+
+func NewAwsIntegrationIAMPermissionsResourceCollectionDataSource() datasource.DataSource {
+	return &awsIntegrationIAMPermissionsResourceCollectionDataSource{}
+}
+
+type awsIntegrationIAMPermissionsResourceCollectionDataSourceModel struct {
+	ID                               types.String `tfsdk:"id"`
+	IAMPermissionsResourceCollection types.List   `tfsdk:"iam_permissions"`
+}
+
+type awsIntegrationIAMPermissionsResourceCollectionDataSource struct {
+	Api  *datadogV2.AWSIntegrationApi
+	Auth context.Context
+}
+
+func (r *awsIntegrationIAMPermissionsResourceCollectionDataSource) Configure(_ context.Context, request datasource.ConfigureRequest, response *datasource.ConfigureResponse) {
+	providerData, _ := request.ProviderData.(*FrameworkProvider)
+	r.Api = providerData.DatadogApiInstances.GetAWSIntegrationApiV2()
+	r.Auth = providerData.Auth
+}
+
+func (d *awsIntegrationIAMPermissionsResourceCollectionDataSource) Metadata(_ context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = "integration_aws_iam_permissions_ResourceCollection"
```

Comment:
💭  : the mixed casing on the name is kind of unusual -- is there a reason it can't be `integration_aws_iam_permissions_resource_collection` ?

Reply:
oops good catch - https://github.com/DataDog/terraform-provider-datadog/pull/3265/commits/62c20c1a4c55e923721df8dc51d1e98b3efe2d7b